src/v/storage/index_state.cc

```diff
@@ -16,7 +16,13 @@
 #include "hashing/crc32c.h"
 #include "model/timestamp.h"
 #include "reflection/adl.h"
-#include "serde/serde.h"
```

Comment:
`serde/serde.h` contains the kitchen sink. removing it causes many cascading changes in dependents.